Application/mqttconnect.py

```diff
@@ -0,0 +1,129 @@
+                         service_water_counter, service_voltage_max, service_voltage_min, service_current_max, service_current_min]
+    service_titles = ['Time', 'Status', 'Water (ml)', 'Countdown Timer',
+                      'Water Counter', 'Voltage Max', 'Voltage Min', 'Current Max', 'Current Min']
+    global_list['service_info'] = dict(zip(service_titles, service_info_list))
+
+    # Gateway info
+    num_of_gateways = len(themsg["metadata"]["gateways"])
+    gateway_info_list = []
+    connection_info_list = []
+    gateway_titles = ['Gateway ID', 'Gateway Latitude',
+                      'Gateway Longitude', 'Gateway Altitude', 'Gateway Location']
+    connection_titles = ['RSSI', 'SNR']
+    for gtw_number in range(num_of_gateways):
+        gtw_id = themsg["metadata"]["gateways"][gtw_number]["gtw_id"]
+        if gtw_id != None:
+            requested_info = requests.get(gateway_link.format(gtw_id))
+            gateway_location = requested_info.json()
+            for each in gateway_location.values():
+                if each.get('location', 0) == 0:
+                    gtw_latitude = 0.0
+                    gtw_longitude = 0.0
+                    gtw_altitude = 0.0
+                    gateway_info_list.append([gtw_id, gtw_latitude,
+                                              gtw_longitude, gtw_altitude, (gtw_latitude, gtw_longitude)])
+                else:
+                    gtw_latitude = each['location'].get('latitude', 0.0)
+                    gtw_longitude = each['location'].get('longitude', 0.0)
+                    gtw_altitude = each['location'].get('altitude', 0.0)
+                    gateway_info_list.append([gtw_id, gtw_latitude,
+                                              gtw_longitude, gtw_altitude, (gtw_latitude, gtw_longitude)])
+        global_list['gateway_info'][gtw_number] = dict(
+            zip(gateway_titles, gateway_info_list[gtw_number]))
+
+        # Connection info
+        connection_rssi = themsg["metadata"]["gateways"][gtw_number]["rssi"]
+        connection_snr = themsg["metadata"]["gateways"][gtw_number]["snr"]
+        connection_info_list.append([connection_rssi, connection_snr])
+        global_list['connection_info'][gtw_number] = dict(
+            zip(connection_titles, connection_info_list[gtw_number]))
+
+    print(
+        f"number of gateways: {num_of_gateways}, \ndevice information: {device_info}, \nservice information: {service_info}, \ngateway information: {gateway_info}, \nconnection information: {connection_info}")
+
+
+def get_data():
+    '''returns the globals for views.py'''
+    return [device_info, service_info, gateway_info, connection_info]
+
+
+def start(username, password, broker, topic):
+    global TTN_USERNAME, TTN_PASSWORD, THE_BROKER, THE_TOPIC
+    TTN_USERNAME = username
+    TTN_PASSWORD = password
+    THE_BROKER = broker
+    THE_TOPIC = topic
+    try:
```

Comment:
Not so clear to me why you have those values as global. If a variable is used only loccally, try to limits its scope because 1) avoid confusion for people trying to undertand ur code from where this is used. 2) less memory consumption ur program would eventually have 3) less error prone(e.g. if in another method u used THE_TOPIC and it somehow refered to that value instaead of ur value that u forgot to define for some reason)